oak-core/src/main/java/org/apache/jackrabbit/oak/query/QueryEngineSettingsService.java

```diff
@@ -77,6 +77,14 @@
                         "the queryPaths of the index is taken into account."
         )
         String getStrictPathRestrictionsForIndexes() default DISABLED_STRICT_PATH_RESTRICTION;
+        
+        @AttributeDefinition(
+                name="Fully qualified class names to ignore when finding caller",
+                description="If non-empty the query engine logs the query statement plus the java package "
```

Comment:
if you prefer not to add the notnull annotation you should explain what a null return value means..... but i would suggest to avoid return null....

Reply:
I have never seen this annotation being used in a class annotated with ```@interface```, as this is normally not instantiated directly, but just created by some OSGI magic. Also the default value makes sure that it is never null.